### ISSUE TITLE
Write unit tests

### ISSUE THREAD

**Issue by Stiivi** [Stiivi/entigen#1]
Reason for not yet having unit tests: "generated output compiles correctly" is sufficient for the time being. However, this might not be very sustainable in the long-run.

**Comment by wizzat** [comment]
You need a couple of "end to end" unit tests for the purposes of describing what the input and outputs look like.

**Comment by Stiivi** [comment]
@wizzat agreed, will add them gradually as the idea stabilizes. This is still proof-of-concept/prototype. 

Just to get an idea how the current Python output looks like for the included example, here is the [generated file](https://gist.github.com/Stiivi/6d5f9ec1b48baf3e70656ecb4ec3df13).

Might be not a bad idea to add small shell script that would:

1. generate few Python outputs
2. run the Python outputs through `mypy` and maybe some additional linters

**Comment by wizzat** [comment]
Sounds like a solid idea - the faster you get something like that in place the less painful it will be to fix it when it invariably breaks.
